replica/table.cc

```diff
@@ -2035,6 +2070,29 @@ future<> table::generate_and_propagate_view_updates(shared_ptr<db::view::view_up
         }
         tracing::trace(tr_state, "Generated {} view update mutations", updates->size());
         auto units = seastar::consume_units(*_config.view_update_concurrency_semaphore, memory_usage_of(*updates));
+
+        // To prevent overload we sleep for a moment before sending another batch of view updates.
+        // The amount of time to sleep for is chosen based on how full the view update backlog is,
+        // the more full the queue of pending view updates is the more aggresively we should delay
+        // new ones.
+        // The first batch of updates doesn't have any delays becase it's slowed down by the other throttling mechanism,
+        // the one which limits the number of incoming client requests by delaying the response to the client.
+        try {
+            if (batch_num > 0) {
+                std::chrono::microseconds throttle_delay =
+                    calculate_throttling_delay_for_view_update_batch(*updates, base_token, gen, timeout);
```

Comment:
We already calculate the endpoints the views have to be sent to in `calculate_throttling_delay_for_view_update_batch()`. I don't know how expensive that is, but maybe it is worth trying to avoid duplicating this effort, by calculating once and then pass it to both `calculate_throttling_delay_for_view_update_batch()` and `mutate_MV()`.

Reply:
The problem is that the new calculate_throttling_delay_for_view_update_batch() calculates the endpoints for 100 different updates. I think in the general (and common) case it will just look at all the nodes in the cluster, in a rather inefficient way ;-)

Reply:
Eh I guess it'll most often look at the whole cluster, but is it that inefficient?

There are cases where the view updates are sent to a few nodes at most (e.g local secondary index clustering range delete), In such cases I don't want to involve the backlogs of other nodes in the throttling calculation, as it'd be wasteful.